lxd/device/nic_routed.go

```diff
@@ -205,6 +206,43 @@ func (d *nicRouted) validateEnvironment() error {
 	return nil
 }
 
+func (d *nicRouted) checkIPAvailability(parent string) error {
+	var addresses []net.IP
+	ipv4AddrStr, ok := d.config["ipv4.address"]
+	if ok {
+		ipv4Addrs := util.SplitNTrimSpace(ipv4AddrStr, ",", -1, true)
+		for _, addr := range ipv4Addrs {
+			addresses = append(addresses, net.ParseIP(addr))
+		}
+	}
+
+	ipv6AddrStr, ok := d.config["ipv6.address"]
+	if ok {
+		ipv6Addrs := util.SplitNTrimSpace(ipv6AddrStr, ",", -1, true)
+		for _, addr := range ipv6Addrs {
+			addresses = append(addresses, net.ParseIP(addr))
+		}
+	}
+
+	errs := make(chan error)
```

Comment:
Can you do `errs := make(chan error, len(addresses)` so you don't need to keep reading from it for the goroutines to complete.